src/structs/pdb.rs

```diff
@@ -1051,6 +1051,26 @@ impl<'a> PDB {
         }
         chains
     }
+
+    /// Returns a vector of unique residue names present in the PDB file.
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - A reference to the PDB struct.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<String>` - A vector of unique residue names.
+    pub fn residues_list(&self) -> Vec<String> {
+        let mut resnames = Vec::new();
+        for residue in self.residues() {
+            let resname = residue.name().unwrap().to_owned();
+            if !resnames.contains(&resname) {
```

Comment:
Here because the number of unique residues could be getting quite high maybe it is worth it to look at a slightly faster way of writing this. If you do [`binary_search`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.binary_search) you can see quicker if the element is in the list, but the list has to be sorted, which can be done by inserting any new element in sorted order instead of pushing it to the end. Luckily the `binary_search` already gives the correct index if the element is not found to be able to insert it in the correct order.

Reply:
good point!